lib/internal/Magento/Framework/Logger/Handler/Base.php

```diff
@@ -31,23 +31,50 @@ class Base extends StreamHandler
     /**
      * @param DriverInterface $filesystem
      * @param string $filePath
+     * @param string $fileName
      */
     public function __construct(
         DriverInterface $filesystem,
-        $filePath = null
+        $filePath = null,
+        $fileName = null
     ) {
         $this->filesystem = $filesystem;
+        if (!empty($fileName)) {
+            $this->fileName = $this->sanitizeFileName($fileName);
+        }
         parent::__construct(
-            $filePath ? $filePath . $this->fileName : BP . $this->fileName,
+            $filePath ? $filePath . $this->fileName : BP . DIRECTORY_SEPARATOR . $this->fileName,
             $this->loggerType
         );
+        
         $this->setFormatter(new LineFormatter(null, null, true));
     }
 
     /**
+     * @param $fileName
+     *
+     * @return string
+     * @throws \InvalidArgumentException
+     */
+    public function sanitizeFileName($fileName)
```

Comment:
I agree with you... actually I prefer making methods protected, in case someone needs to overwrite the class but doesn't want to change the behaviour of this method.

I'm not sure how to make the proper test then, should those be integration test instead of unit tests? What do you think would be the proper approach?

Thanks!